packages/terra-hyperlink/src/Hyperlink.jsx

```diff
@@ -0,0 +1,191 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './Hyperlink.scss';
+
+const cx = classNames.bind(styles);
+
+const KEYCODES = {
+  ENTER: 13,
+  SPACE: 32,
+  TAB: 9,
+};
+
+const variants = {
+  DEFAULT: 'default',
+  EXTERNAL: 'external',
+  IMAGE: 'image',
+  VIDEO: 'video',
+  AUDIO: 'audio',
+  DOCUMENT: 'document',
+};
+
+const propTypes = {
+  /**
+   * The content to display inside link.
+   */
+  children: PropTypes.node,
+  /**
+   * Sets the href of the link.
+   */
+  href: PropTypes.string,
+  /**
+   * Whether or not the link should be disabled.
+   */
+  isDisabled: PropTypes.bool,
+  /**
+   * Whether or not the link should display and underline by default.
+   */
+  isUnderlineHidden: PropTypes.bool,
+  /**
+   * Callback function triggered when clicked.
+   */
+  onClick: PropTypes.func,
+  /**
+   * Callback function triggered when hyperlink loses focus.
+   */
+  onBlur: PropTypes.func,
+  /**
+   * Callback function triggered when hyperlink gains focus.
+   */
+  onFocus: PropTypes.func,
+  /**
+   * Callback function triggered when key is pressed.
+   */
+  onKeyDown: PropTypes.func,
+  /**
+   * Callback function triggered when key is released.
+   */
+  onKeyUp: PropTypes.func,
+  /**
+   * Sets the hyperlink variant. One of `default`, `external`, `image`, `video`, `audio`, `document`.
+   */
+  variant: PropTypes.oneOf(['default', 'external', 'image', 'video', 'audio', 'document']),
+};
+
+const defaultProps = {
+  isDisabled: false,
+  variant: variants.DEFAULT,
+};
+
+class Hyperlink extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { active: false, focused: false, mouseWasClicked: false };
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.handleKeyUp = this.handleKeyUp.bind(this);
+    this.handleOnBlur = this.handleOnBlur.bind(this);
+  }
+
+  handleOnBlur(event) {
+    this.setState({ focused: false });
+
+    if (this.props.onBlur) {
+      this.props.onBlur(event);
+    }
+  }
+
+  handleKeyDown(event) {
+    // Add active state to FF browsers
+    if (event.nativeEvent.keyCode === KEYCODES.SPACE) {
+      this.setState({ active: true });
+
+      // Follow href on space keydown when rendered as an anchor tag
+      if (this.props.href) {
+        // Prevent window scrolling
+        event.preventDefault();
+        window.location.href = this.props.href;
```

Comment:
What if they specified target _blank on the html attribute? We should choose between window.location and window.open depending on what they chose.

Reply:
Removed this logic altogether, it was copy pasta from our Button component that isn't needed for hyperlinks. https://github.com/cerner/terra-core/pull/1562/commits/5d1a6de0b4dd80fbbb620354ee5b823764386a5b